projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -4168,18 +4178,31 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
                     let searchIndex = searchValue.indexOf(searchText);
                     const pageIndex = this.paging ? Math.floor(i / this.perPage) : 0;
 
-                    while (searchIndex !== -1) {
-                        this.lastSearchInfo.matchInfoCache.push({
-                            row: rowIndex,
-                            column: j,
-                            page: pageIndex,
-                            index: occurenceIndex++,
-                            groupByRecord: groupByRecord,
-                            item: dataRow
-                        });
-
-                        searchValue = searchValue.substring(searchIndex + searchText.length);
-                        searchIndex = searchValue.indexOf(searchText);
+                    if (exactMatch) {
+                        if (searchValue === searchText) {
+                            this.lastSearchInfo.matchInfoCache.push({
+                                row: rowIndex,
+                                column: j,
+                                page: pageIndex,
+                                index: occurenceIndex++,
+                                groupByRecord: groupByRecord,
+                                item: dataRow
+                            });
+                        }
+                    } else {
+                        while (searchIndex !== -1) {
```

Comment:
Move the initiation of the searchIndex and occurrenceIndex variables in this else block.